src/uct/ib/rc/accel/rc_mlx5_iface.c

```diff
@@ -196,17 +190,28 @@ uct_rc_mlx5_iface_handle_failure(uct_ib_iface_t *ib_iface, void *arg,
     uct_rc_mlx5_ep_t   *ep     = ucs_derived_of(uct_rc_iface_lookup_ep(iface,
                                                                        qp_num),
                                                 uct_rc_mlx5_ep_t);
+    uint16_t           pi      = ntohs(cqe->wqe_counter);
     ucs_log_level_t    log_lvl;
     uct_ib_mlx5_txwq_t txwq_copy;
     size_t             txwq_size;
-    ucs_status_t       err_handler_status;
+    ucs_status_t       status;
 
     if (!ep) {
         return;
     }
 
+    uct_rc_mlx5_common_update_tx_res(iface, &ep->tx.wq, &ep->super.txqp, pi);
+    uct_rc_txqp_purge_outstanding(iface, &ep->super.txqp, ep_status, pi, 0);
+
+    if (ep->super.flags & (UCT_RC_EP_FLAG_NO_ERR_HANDLER |
+                           UCT_RC_EP_FLAG_FLUSH_CANCEL)) {
+        return;
+    }
+
+    ep->super.flags |= UCT_RC_EP_FLAG_NO_ERR_HANDLER;
+
     /* Create a copy of RC txwq for completion error reporting, since the QP
-     * would be released by set_ep_failed()*/
+     * may be released by error handler */
     txwq_copy = ep->tx.wq;
     txwq_size = UCS_PTR_BYTE_DIFF(ep->tx.wq.qstart, ep->tx.wq.qend);
     txwq_copy.qstart = ucs_malloc(txwq_size, "rc_txwq_copy");
```

Comment:
I thought we can avoid this now

Reply:
the EP should be flushed (with cancel in case of err), then destroyed after receiving all the completions. do I miss something?

Reply:
@yosefe could you please help here.
Why should we allow ep destroy from err_handler?
Is it required?

Reply:
we should not allow ep_destroy from err handler
UCP should not do this today
if you see otherwise - can you pls post the stacktrace?

Reply:
I just see test test_uct_peer_failure_cb.desproy_ep_cb which explicitly tests it.
If it's obsolete - may remove it, no need to save bool no_dci, txwq_copy, etc...